sklearn/metrics/pairwise.py

```diff
@@ -1930,13 +1939,13 @@ def pairwise_distances(
 
     Parameters
     ----------
-    X : ndarray of shape (n_samples_X, n_samples_X) or \
+    X : array-like of shape (n_samples_X, n_samples_X) or \
```

Comment:
Since we call `X.dtype` without any other validation on `X`, I really think that we need an `ndarray` and not an `array-like`.
The same for `Y`.